src/java/org/apache/cassandra/streaming/StreamManager.java

```diff
@@ -260,10 +269,20 @@ public StreamManager()
         logger.info("Storing streaming state for {} or for {} elements", duration, numElements);
         states = CacheBuilder.newBuilder()
                              .expireAfterWrite(duration.quantity(), duration.unit())
-                             .maximumSize(numElements)
+                             .maximumWeight(10 * 1024 *1024)
+                             .weigher(new streamingStateWeigher())
                              .build();
     }
 
+    private class streamingStateWeigher implements Weigher<TimeUUID,StreamingState>
```

Comment:
classes should be upper case for the first letter in the name.

Also you should use `static` when you don't need access to `StreamManager.this`